src/System.Net.Http/tests/FunctionalTests/ResponseStreamTest.cs

```diff
@@ -68,6 +69,26 @@ public ResponseStreamTest(ITestOutputHelper output)
             }
         }
 
+        [Fact]
+        public async Task ReadAsStreamAsync_Cancel_TaskIsCanceled()
+        {
+            var cts = new CancellationTokenSource();
+
+            using (var client = new HttpClient())
+            {
+                HttpResponseMessage response =
+                    await client.GetAsync(HttpTestServers.RemoteEchoServer, HttpCompletionOption.ResponseHeadersRead);
+                Stream stream = await response.Content.ReadAsStreamAsync();
+
+                var buffer = new byte[2048];
+                Task task = stream.ReadAsync(buffer, 0, buffer.Length, cts.Token);
```

Comment:
> But I thought it was more valuable to have an HttpClientHandler test to help exercise CurlHandler as well.

Agreed :smile:

> I could modify the check and validate that the task moves to either a Canceled or Completed (successfully) state

That seems reasonable.  If the goal of the test is mainly to verify that it doesn't hang, then simply verifying that it completes successfully or due to cancellation sounds good.

Do we also need to dispose of the stream and/or response?